watershed-android/watershed/app/src/main/java/com/blueprint/watershed/Tasks/TaskFragment.java

```diff
@@ -161,36 +172,90 @@ public void onDetach() {
      * depending on what tab is being clicked on.
      */
     private void getTasksRequest(){
+        Log.i("watasdfasdf", String.valueOf(mArgs.getInt(OPTION)));
         HashMap<String, JSONObject> params = new HashMap<String, JSONObject>();
 
         TaskListRequest taskListRequest = new TaskListRequest(getActivity(), params, new Response.Listener<ArrayList<Task>>() {
             @Override
             public void onResponse(ArrayList<Task> tasks) {
-                setTasks(tasks);
-                mParentActivity.getSpinner().setVisibility(View.GONE);
-                mAllTaskAdapter.notifyDataSetChanged();
-                mUserTaskAdapter.notifyDataSetChanged();
+                if (mArgs.getInt(OPTION) == USER) {
+                    tasks = getUserTasks(tasks);
+                    if (tasks.size() > 0) {
+                        showList();
+                        setUserTasks(tasks);
+                        mUserTaskAdapter.notifyDataSetChanged();
+                        for(int i=0; i < mUserTaskAdapter.getGroupCount(); i++) mListView.expandGroup(i);
+                    } else {
+                        hideList();
+                    }
+                } else {
+                    if (tasks.size() > 0) {
+                        showList();
+                        setAllTasks(tasks);
+                        mAllTaskAdapter.notifyDataSetChanged();
+                        for(int i=0; i < mAllTaskAdapter.getGroupCount(); i++) mListView.expandGroup(i);
+                    } else {
+                        hideList();
+                    }
+                }
+
                 if (mSwipeLayout != null) mSwipeLayout.setRefreshing(false);
             }
-        });
+        }, mSwipeLayout);
```

Comment:
it sends the layout so even if the request errors out it doesn't get refreshed 4ever